### ISSUE TITLE
dataset import does not work sometimes

### ISSUE THREAD

**Issue by demo125** [doccano/doccano#1377]
https://github.com/doccano/doccano/blob/7a87d7a36efb2bfbe0aba919f91413bb091742b1/backend/api/tasks.py#L67

This one line took me one day of finding why data import SOMETIMES didnt work. Importing was successfull but SOME documents were linked to wrong annotations. Obviously, this is not good practice and the code should look like this:

<codde>
#  now = datetime.datetime.now() <br>
documents = self.data_class.objects.bulk_create(dataset) <br>
#  ids = self.data_class.objects.filter(created_at__gte=now)  <br>
return documents
</code>

**Comment by Hironsan** [comment]
This code does not work properly except for PostgreSQL and MariaDB 10.5+.
- [Django bulk create](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-create)

```python
# now = datetime.datetime.now()
documents = self.data_class.objects.bulk_create(dataset)
# ids = self.data_class.objects.filter(created_at__gte=now)
return documents
```

**Comment by henrytseng** [comment]
I believe I'm receiving the same issue.  Using postgres

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/celery/app/trace.py", line 450, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/celery/app/trace.py", line 731, in __protected_call__
    return self.run(*args, **kwargs)
  File "/doccano/backend/api/tasks.py", line 120, in injest_data
    factory.create(buffer.data, user, project)
  File "/doccano/backend/api/tasks.py", line 84, in create
    ids = self.create_data(examples, project)
  File "/doccano/backend/api/tasks.py", line 66, in create_data
    self.data_class.objects.bulk_create(dataset)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 514, in bulk_create
    returned_columns = self._batched_insert(
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 1287, in _batched_insert
    inserted_rows.extend(self._insert(
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 1270, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1416, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.DataError: value too long for type character varying(100)
```

**Comment by demo125** [comment]
henrytseng Its another bug. Just got the same error while I was trying to import two absolutely identical jsonl files. One works and other not. The problem was in the name of the second file. It can not be too long - at most 36 characters.
document content:
{ "data": "foo"}
abcdefghijklmnoprstuvwxyz12345.jsonl => error
abcdefghijklmnoprstuvwxyz1234.jsonl => works

**Comment by henrytseng** [comment]
Yeah, I noticed that last night and made the DB correction in my own setup.

**Comment by Vydruth** [comment]
I face the same issue. my file name is not over 36 charecters. I've tried different file types for uploading datasets, it works sometimes it fails sometimes..

**Backend Logs:**
```
Internal Server Error: /v1/projects/251/upload
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.8/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/doccano/backend/api/views/import_dataset.py", line 35, in post
    tus = [TemporaryUpload.objects.get(upload_id=upload_id) for upload_id in upload_ids]
  File "/doccano/backend/api/views/import_dataset.py", line 35, in <listcomp>
    tus = [TemporaryUpload.objects.get(upload_id=upload_id) for upload_id in upload_ids]
  File "/usr/local/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 435, in get
    raise self.model.DoesNotExist(
django_drf_filepond.models.TemporaryUpload.DoesNotExist: TemporaryUpload matching query does not exist.
```

**Frontend Network logs on Chrome:**
<img width="1311" alt="Screensnip-1" src="https://user-images.githubusercontent.com/10277192/128949647-97b7b574-6d91-48eb-ad40-4acbbce492ae.png">

<img width="1664" alt="Screensnip-2" src="https://user-images.githubusercontent.com/10277192/128949595-c46005c5-07a0-44d4-ad03-e11b1e8b6dd0.png">

**Comment by colinux** [comment]
Just had this issue yesterday. The `api_example#filename` column contains an _absolute_ path which can be very long (at least on heroku setup), and can exceed the 100 chars limits.

For example I found `/doccano/backend/media/Bhfxxxxxxxxxxxxxxxxxxxxx/ABoDxxxxxxxxxxxxxxxxxxxx/10123-dataset.jsonl` which has a length of 92. With a slightly longer filename, the absolute file path would easily exceed 100 chars.

Resolved by increasing the limit from 100 to 150 : 
`alter table api_example alter column filename type character varying(150);`
